pkg/controller/hibernation/aws_actuator.go

```diff
@@ -149,7 +275,32 @@ func getAWSClient(cd *hivev1.ClusterDeployment, c client.Client, logger log.Fiel
 	return awsclient.New(c, options)
 }
 
-func getClusterInstanceIDs(cd *hivev1.ClusterDeployment, c awsclient.Client, states sets.String, logger log.FieldLogger) ([]*string, error) {
+// filterOutSpotInstances removes the spot instances from the list and returns it. It
+// also returned the spot instances that were filtered out in a separate list.
+func filterOutSpotInstances(instances []*ec2.Instance) ([]*ec2.Instance, []*ec2.Instance) {
+	var spotInstances []*ec2.Instance
+	n := 0
+	for _, i := range instances {
+		if aws.StringValue(i.InstanceLifecycle) == "spot" {
+			spotInstances = append(spotInstances, i)
+			continue
+		}
+		instances[n] = i
+		n++
+	}
+	instances = instances[:n]
+	return instances, spotInstances
+}
+
+func instanceIDs(instances []*ec2.Instance) []string {
+	result := make([]string, len(instances))
+	for idx, i := range instances {
+		result[idx] = aws.StringValue(i.InstanceId)
```

Comment:
Why is `aws.StringValue()` needed here? Are we worried about the backing list changing under us? Keeping the IDs as pointers (making this func return `[]*string`) avoids the need to use e.g. `aws.StringSlice()` elsewhere. (And it saves a teeny bit of memory.)

[Later] Okay, I see that sometimes we're addressing and sometimes dereferencing. So meh, either way. But still a slight preference for keeping the pointers as much as possible.

Reply:
the list of instance ID strings is useful for reporting/logging the work in action. pointers of string is not useful anywhere except when calling AWS API, and the SDK recognizes that by provding aws.StringSlice function.

my preference will be to keep a list of strings vs pointers to them..